src/components/MiddleBar.js

```diff
@@ -0,0 +1,60 @@
+import React, { useState } from 'react';
+import { Link } from 'react-router-dom';
+import { useSelector } from 'react-redux';
+import pokeball100 from '../images/pokeball-100.png';
+
+function Header() {
+  const pokemonList = useSelector((state) => state.preview);
+  const [focus, setFocus] = useState(false);
+  const [input, setInput] = useState('');
+  const [search, setSearch] = useState('');
+
+  const filter = (value) => {
+    const list = [...pokemonList];
+    const filteredData = list.filter((el) => ((el.name.includes(value) || el.id.includes(value))));
+    if (filteredData.length) setSearch(value);
+    if (filteredData.length === 1) setSearch(filteredData[0].id);
+    if (filteredData.length) setSearch(value);
+    if (filteredData.length === 1) setSearch(filteredData[0].id);
+  };
+
+  const inputHandler = (e) => {
+    const value = e.target.value.toLowerCase();
+    setInput(value);
+    filter(value);
+  };
+  const focusHandler = () => {
+    setFocus(true);
+  };
+
+  return (
+    <>
+      <section className="middle-bar">
+        <h3>Pokemons</h3>
+        <div className="search">
+          <input
+            className="search-bar"
+            type="text"
+            placeholder="Search a Pokemon"
+            onChange={inputHandler}
+            onFocus={focusHandler}
+            value={input}
+            required
+          />
+          {focus && input ? (
+            <span className="searchResults">
+              {
+                <Link to={`/details/${search}`}>
+                  <div>
+                    <img src={pokeball100} alt="" className="search-ball" />
+                  </div>
+                </Link> || null
+              }
+            </span>
+          ) : null}
+        </div>
+      </section>
```

Comment:
- [ OPTIONAL ] I think the search bar is not working well so kindly fix the problem to be able to filter the cards according to the requirements.

![image](https://user-images.githubusercontent.com/58553711/188037929-1122fa29-5b54-4bfe-8d03-03fa7f1b0b9b.png)